File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/CountEqualElement/CountEqualElement_test.js
import assert from 'assert'
import solution from './CountEqualElement.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const nums = [2, 5, 6, 3, 2, 6, 6]

    // when
    const result = solution(nums)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const nums = [8, 5, 10, 10, 10, 6, 10, 9, 5, 6]

    // when
    const result = solution(nums)

    // then
    assert.deepStrictEqual(8, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const nums = [100]

    // when
    const result = solution(nums)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const nums = []

    // when
    const result = solution(nums)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const nums = [100, 100, 100]

    // when
    const result = solution(nums)

    // then
    assert.deepStrictEqual(3, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const nums = [7, 8, 5, 10, 6, 5, 10, 7, 9, 5]

    // when
    const result = solution(nums)

    // then
    assert.deepStrictEqual(5, result)
    done();
  });
})

File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/CountEqualElement/CountEqualElement_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.CountEqualElement.CountEqualElement import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        nums = [2, 5, 6, 3, 2, 6, 6]

        # when
        result = solution(nums)

        # then
        self.assertEqual(4, result)

    def test_2(self):
        # given
        nums = [8, 5, 10, 10, 10, 6, 10, 9, 5, 6]

        # when
        result = solution(nums)

        # then
        self.assertEqual(8, result)

    def test_3(self):
        # given
        nums = [100]

        # when
        result = solution(nums)

        # then
        self.assertEqual(0, result)

    def test_4(self):
        # given
        nums = []

        # when
        result = solution(nums)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        nums = [100, 100, 100]

        # when
        result = solution(nums)

        # then
        self.assertEqual(3, result)

    def test_6(self):
        # given
        nums = [7, 8, 5, 10, 6, 5, 10, 7, 9, 5]

        # when
        result = solution(nums)

        # then
        self.assertEqual(5, result)


if __name__ == '__main__':
    unittest.main()
